nautobot/core/templatetags/helpers.py

```diff
@@ -756,7 +756,9 @@ def saved_view_modal(
             if current_saved_view_pk:
                 current_saved_view_pk = current_saved_view_pk[0]
                 try:
-                    current_saved_view = SavedView.objects.restrict(request.user, "view").get(pk=current_saved_view_pk)
+                    # We are not using .restrict(request.user, "view") here
+                    # User should be able to see any saved view that he has the list view access to.
+                    current_saved_view = SavedView.objects.get(pk=current_saved_view_pk)
```

Comment:
This may need to be revisited when we implement #5471 - I'm guessing at that point we should make it so that a non-staff user can view their own saved views, and those marked as "shared", but not any others?

Might make sense to create a helper function specifically for saved-view queryset restriction, as it may need to be used in quite a few places.